viewer-prototype/src/browser/trace-viewer/components/utils/filtrer-tree/tree-node.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faChevronDown, faChevronRight } from '@fortawesome/free-solid-svg-icons'
+import { CheckboxComponent } from '../checkbox-component';
+import { Indent } from './indent';
+
+export type TreeNode = {
+    id: number;
+    parentId: number;
+    name: string;
+    isLeaf: boolean;
+    expanded: boolean;
+    children: [];
+    isRoot: boolean;
+}
+
+type TreeNodeComponentProps = {
+    node: TreeNode;
+    level: number;
+    isCheckable: boolean;
+    checkedSeries: number[];
+    getChildNodes: (node: TreeNode) => TreeNode[];
```

Comment:
Do you need the getChildNodes() method or can you just use node.children whenever required?